### PR TITLE
torch: add DistributedDataParallel module

### PR DESCRIPTION
**Pull request by pleasantrabbit**
Add DistributedDataParallel module for PyTorch. Currently it's
compatible with the major use case of PyTorch DDP, i.e. single-process
single-GPU mode.

Signed-off-by: Yulu Jia <yulu.jia@bytedance.com>

### CODE REVIEW

File: example/pytorch/main_bps_ddp.py [hunk 1]
```diff
@@ -0,0 +1,420 @@
+import argparse
```
[the remark this answers is not in the document]
Comment (pleasantrabbit) [reply example/pytorch/main_bps_ddp.py:1]: I will rename it. 
> Is this modified from an existing code somewhere?
yes it's modified from an official PyTorch example. I will add a link to the original code.

File: docs/DistributedDataParallel.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+```python
+# ddp_example.py
+from byteps.torch.parallel import DistributedDataParallel
+
+model = DistributedDataParallel(model, device_ids=[i])
+output = model(data)
+loss = F.nll_loss(output, target)
+loss.backward()
+model.synchronize()
```
[the remark this answers is not in the document]
Comment (pleasantrabbit) [reply docs/DistributedDataParallel.md:34]: in the case described on line 22, model.synchronize() is required. I will add a comment in the code snippet to make it clearer.
